### PR TITLE
Add new error code mechanism, update error usage in pkg

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Add new error code mechanism according to https://github.com/pingcap/tidb/blob/master/docs/design/2020-05-08-standardize-error-codes-and-messages.md

This is the first part of https://github.com/pingcap/ticdc/issues/806

### What is changed and how it works?

- Replace existing defined error in TiCDC to new error
- Replace all errors used in `pkg` to new error

### Check List 

Tests 

 - Unit test
 - Integration test
 
### Release note

- Add new error code mechanism based on TiDB error code RFC

### CODE REVIEW

File: pkg/errors/helper.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/pingcap/errors"
+)
+
+// WrapError generates a new error based on given `*errors.Error`, wraps the err
+// as parameter via the `GenWithStackByArgs`
+// The message format of `*errors.Error` must be `format message: %s`
+// In this way the Error() string of returned error will keep the RFC format
+// In the future if the error lib provides a native error wrap way, we could just
+// change this function to the new API.
+func WrapError(rfcError *errors.Error, err error) error {
```
[the remark this answers is not in the document]
Comment (nolouch) [reply pkg/errors/helper.go:26]: agree, we have the same problem. in addition,  it's better to compatible with [GO2 Proposal](https://go.googlesource.com/proposal/+/master/design/go2draft-error-values-overview.md)

[the remark this answers is not in the document]
Comment (overvenus) [reply pkg/errors/helper.go:24]: Do we still need it?

**Review by overvenus — COMMENTED** [review]
Rest LGTM, please fix the confliction.

File: pkg/errors/helper.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (amyangfei) [reply pkg/errors/helper.go:24]: Still need it, for two reasons
- we can wrap a nil error, it returns nil in this wrap function, while in the `pingcap/errors` wrap a nil error still returns a new error, but error message is <nil>
- It is easy to control the error message, though we use `GenWithStackByCause` now

[the remark this answers is not in the document]
Comment (overvenus) [reply pkg/errors/helper.go:24]: > we can wrap a nil error, it returns nil in this wrap function, while in the pingcap/errors wrap a nil error still returns a new error, but error message is <nil>

Sounds like a bug 🤔

**Pull request merged**